osbenchmark/builder/installers/docker_installer.py

```diff
@@ -0,0 +1,169 @@
+
+    def _get_config_vars(self, node):
+        provisioner_defaults = {
+            "cluster_name": self.provision_config_instance.variables["cluster_name"],
+            "node_name": node.name,
+            # we bind-mount the directories below on the host to these ones.
+            "install_root_path": "/usr/share/opensearch",
+            "data_paths": ["/usr/share/opensearch/data"],
+            "log_path": "/var/log/opensearch",
+            "heap_dump_path": "/usr/share/opensearch/heapdump",
+            # Docker container needs to expose service on external interfaces
+            "network_host": "0.0.0.0",
+            "discovery_type": "single-node",
+            "http_port": str(node.port),
+            "transport_port": str(node.port + 100),
+            "cluster_settings": {}
+        }
+
+        config_vars = {}
+        config_vars.update(self.provision_config_instance.variables["origin"]["docker"])
+        config_vars.update(provisioner_defaults)
+
+        return config_vars
+
+    def _get_docker_vars(self, node, log_dir, heap_dump_dir, mounts):
+        docker_vars = {
+            "os_version": self.provision_config_instance.variables["origin"]["distribution"]["version"],
+            "docker_image": self.provision_config_instance.variables["origin"]["docker"]["docker_image"],
+            "http_port": node.port,
+            "os_data_dir": node.data_paths[0],
+            "os_log_dir": log_dir,
+            "os_heap_dump_dir": heap_dump_dir,
+            "mounts": mounts
+        }
+        self._add_if_defined_for_provision_config_instance(docker_vars, "docker_mem_limit")
+        self._add_if_defined_for_provision_config_instance(docker_vars, "docker_cpu_count")
+        return docker_vars
+
+    def _add_if_defined_for_provision_config_instance(self, variables, key):
+        if key in self.provision_config_instance.variables["origin"]["docker"]:
+            variables[key] = self.provision_config_instance.variables["origin"]["docker"][key]
+
+    def _render_template_from_docker_file(self, variables):
+        compose_file = os.path.join(paths.benchmark_root(), "resources", "docker-compose.yml.j2")
+        env = jinja2.Environment(loader=jinja2.FileSystemLoader(io.dirname(compose_file)), autoescape=select_autoescape(['html', 'xml']))
+        return self._render_template(env, variables, compose_file)
+
+    def cleanup(self, host):
+        if self.provision_config_instance.variables["preserve_install"]:
+            console.info("Preserving benchmark candidate installation.", logger=self.logger)
+            return
+
+        self.logger.info("Wiping benchmark candidate installation at [%s].", host.node.binary_path)
+
+        for data_path in host.node.data_paths:
+            self._delete_path(host, data_path)
+
+        self._delete_path(host, host.node.binary_path)
+
+    def _delete_path(self, host, path):
+        self.executor.execute(host, "rm -r " + path)
```

Comment:
This seems dangerous to me. Open to other ideas

Reply:
If I use rm -rf, then would do something to check if the path string has some signatures to be OK to be deleted, including
* not empty
* not a very high level folder like "/" or home folder
* not "*"

Reply:
Validating the `path` was one thought I had, but it doesn't seem quite secure enough. May be the best option available though.